shared/components/card.js

```diff
@@ -199,17 +162,21 @@ class Card extends Component {
           date: this.state.date, dataSource: this.props.dataSource, isSaved: this.state.isSaved});
       }}>
         <Image style={styles.CardImage}
-             source = {{uri: rowData}}/>
+               source = {{uri: rowData}}/>
       </TouchableOpacity>
     );
   }
 
   handlePressStar() {
     if (this.state.isSaved === true) {
-      this.setState({isSaved: false});
+      if (this.props.currentScene !== 'myPage') {
```

Comment:
This if statement has to locate in outside of `if (this.state.isSaved === true) {`
Then you can also remove the if statement at line 177